docs/concepts/ipfs-gateway.md

```diff
@@ -3,103 +3,120 @@ title: IPFS Gateway
+- [Authentication support](#authenticated-gateways)
+- [Read support](#read-only-gateways)
 
-_Private gateways_ are configured to limit access to requests from specific domains or parts of the public internet. 
+### Recursive vs. non-recursive gateways
 
-They are frequently, but not exclusively, used behind firewalls. Running [IPFS Desktop](https://github.com/ipfs-shipyard/ipfs-desktop#ipfs-desktop) or another form of IPFS node triggers connection attempts to other IPFS peers. Private network administrators may treat such connection attempts as potential security vulnerabilities. Private IPFS gateway servers located inside the private network and running a trusted code base provide an alternative architecture for read/write access to externally-hosted IPFS content.
+Recursive gateways are gateways that will attempt to retrieve content from other peers on the network if they do not have it locally.
 
-### Public gateways
+Non-recursive gateways are gateways that only serve content that they have themselves. For example, [Kubo](https://github.com/ipfs/kubo) can be configured to act as a non-recursive gateway by setting the [`NoFetch`](https://github.com/ipfs/kubo/blob/master/docs/config.md#gatewaynofetch) option.
 
-For more information about public gateways, see the [Public IPFS Gateways](./public-utilities.md#public-ipfs-gateways)
+In general, recursive gateways are more powerful for end-users because they abstract away all details of the peer-to-peer network. However, they are much more resource-intensive for operators and prone to abuse.
 
-## Gateway types
+Non-recursive gateways are becoming a popular way to provide IPFS content to the network (as an alternative or in addition to Bitswap). Non-recursive provider records can only be announced to the [IPNI](../concepts/ipni/), but not the [DHT](../concepts/dht.md).
 
-There are multiple gateway types, each with specific use case, security, performance, and functional implications.
+## Trusted vs. trustless gateways
 
-- [Read support](#read-only-gateways)
-- [Authentication support](#authenticated-gateways)
-- [Resolution style](#resolution-style)
-- [Service](#gateway-services)
+See [Trusted vs. Trustless Gateways](../reference/http/gateway/#trusted-vs-trustless) for more information.
 
 ### Read-only gateways
 
 _Read-only gateways_ are the simplest kind of gateway. This gateway type provides a way to fetch IPFS content using the HTTP GET method.
 
-### Authenticated gateways
+## Authenticated gateways
 
 If a gateway provider wants to limit access to requests with authentication, they may need to configure a reverse proxy, develop an IPFS plugin, or set a cache-layer above IPFS.
 
 Configuring a reverse proxy is the most popular way for providers handling authentication. Reverse proxy can also keep the original IPFS API calls which makes gateway adaptable to all IPFS SDK and toolkits.
 
 ![Auth with Reverse proxy](./images/ipfs-gateways/public-authed-gateway.png)
```

Comment:
nit: this image is wrong and confusing (gateway has no `/api/v0`), just remove it.

```suggestion
```